src/Apps/Order2/Routes/Checkout/Components/Order2FulfillmentDetailsStep/Order2FulfillmentDetailsStep.tsx

```diff
@@ -46,41 +40,51 @@ export const Order2FulfillmentDetailsStep: React.FC<
       backgroundColor="mono0"
       py={2}
     >
-      {stepStatus === TempStepStatus.COMPLETE ? (
-        <Box px={2}>
-          <Order2FulfillmentDetailsCompletedView
-            fulfillmentDetails={savedFulfillmentDetails}
-            onClickEdit={() => setStepStatus(TempStepStatus.ACTIVE)}
-            fulfillmentOption={orderData?.selectedFulfillmentOption}
-          />
+      <Box
+        px={2}
+        display={stepState === CheckoutStepState.COMPLETED ? "block" : "none"}
+      >
+        <Order2FulfillmentDetailsCompletedView
+          fulfillmentDetails={savedFulfillmentDetails}
+          onClickEdit={() => editFulfillmentDetails()}
+          fulfillmentOption={selectedFulfillmentOption} // TODO: should this come from the context where step is?
+        />
+      </Box>
+      {fulfillmentOptions?.some(option => option.type === "PICKUP") ? (
```

Comment:
Rather than inlining logic like this, better to assign to a self-describing variable within the body of the function. Something like `isPickup`